collect_app/src/main/java/org/odk/collect/android/widgets/WidgetAnswerFactory.kt

```diff
@@ -0,0 +1,52 @@
+package org.odk.collect.android.widgets
+
+import android.content.Context
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.lifecycle.viewmodel.compose.viewModel
+import org.javarosa.core.model.Constants
+import org.javarosa.form.api.FormEntryPrompt
+import org.odk.collect.android.utilities.MediaUtils
+import org.odk.collect.android.utilities.QuestionMediaManager
+import org.odk.collect.android.widgets.video.VideoWidgetAnswer
+import org.odk.collect.android.widgets.video.VideoWidgetAnswerViewModel
+
+@Composable
+fun widgetAnswer(
+    prompt: FormEntryPrompt,
+    context: Context,
+    answer: String?,
+    questionMediaManager: QuestionMediaManager,
+    mediaUtils: MediaUtils
+): (@Composable () -> Unit)? = when (prompt.controlType) {
+    Constants.CONTROL_VIDEO_CAPTURE -> videoWidgetAnswer(context, answer, questionMediaManager, mediaUtils)
+    else -> throw IllegalArgumentException("Unsupported control type: ${prompt.controlType}")
+}
+
+@Composable
+private fun videoWidgetAnswer(
```

Comment:
This could probably just be inlined in `WidgetAnswer`.

Reply:
I will be adding more cases for other question types so I think keeping them in separate methods makes sense.